site/profile/manifests/userportal.pp

```diff
@@ -0,0 +1,121 @@
+class profile::userportal::server (
+  $password
+){
+  $domain_name = lookup('profile::freeipa::base::domain_name')
+  package {['python3-virtualenv', 'python38', 'python38-devel']: }
+  package {['openldap-devel', 'gcc', 'mariadb-devel']: }
+
+  # Using python3.8 with gunicorn
+  exec { 'create virtualenv':
+    command => '/usr/bin/virtualenv-3 --python="/usr/bin/python3.8" /var/www/userportal-env',
+    unless  => '/usr/bin/test -d /var/www/userportal-env',
+    require => [Package['python3-virtualenv'], Package['python38']],
+  }
+
+  file { '/var/www/userportal/':
+    ensure => 'directory',
+    owner  => 'apache',
+    group  => 'apache',
+  }
+  -> vcsrepo { '/var/www/userportal/':
```

Comment:
Is there an interest in maintaining a git repo here or we could use an archive resource and fetch `https://github.com/guilbaults/TrailblazingTurtle/archive/7940ae14891a60d18afd1d9d009dada044512b0f.zip`?